snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/components/MapViewModel.kt

```diff
@@ -201,7 +204,8 @@ class MapViewModel(
             isCreateButtonEnabled.value = true
         }
 
-        if (geometryEditor.selectedElement.value != null) {
+        val selectedElement = geometryEditor.selectedElement.value
+        if (selectedElement != null && selectedElement.canDelete) {
```

Comment:
```suggestion
       if (selectedElement?.canDelete == true) {
```

This checks null and reads easier, potentially